diff --git a/publish/generic-preview/oscal_metadata_html.xsl b/publish/generic-preview/oscal_metadata_html.xsl
index b34a6fd..9b3ec3c 100644
--- a/publish/generic-preview/oscal_metadata_html.xsl
+++ b/publish/generic-preview/oscal_metadata_html.xsl
@@ -249,5 +249,60 @@
       </p>
    </xsl:template>
    
+<!-- TAMUS ADDED CONTENT BELOW HERE -->
+
+  <xsl:template priority="1" match="metadata" mode="metadata">
+      <section class="metadata">
+         <details>
+            <summary><span class="lbl">OSCAL Catalog Metadata</span></summary>
+            <xsl:apply-templates select="published, last-modified, version, oscal-version, doc-id, prop, link"/>
+            <!--
+            <xsl:for-each-group select="role" group-by="true()">
+               <div class="block roles">
+                  <xsl:apply-templates select="current-group()"/>
+               </div>
+            </xsl:for-each-group>
+            <xsl:for-each-group select="party" group-by="true()">
+               <div class="block parties">
+                  <xsl:apply-templates select="current-group()"/>
+               </div>
+            </xsl:for-each-group>
+            -->
+            <!-- responsibile-parties elements bind parties to roles, handled with parties -->
+            <xsl:apply-templates select="remarks"/>
+         </details>
+      </section>
+   </xsl:template>
+
+   <xsl:template priority="1" match="published">
+      <p><span class="inline-head">Published Date</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="."/>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="1" match="last-modified">
+      <p><span class="inline-head">Last Substantively Modified Date</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="."/>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="1" match="version">
+      <p><span class="inline-head">Version</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="."/>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="1" match="oscal-version"/>
+
+   <xsl:template priority="1" match="role"/>
+
+   <xsl:template priority="1" match="party"/>
+
+   <xsl:template priority="1" match="prop[@name = 'resolution-tool']"/>
    
+   <xsl:template priority="1" match="link[@rel = 'source-profile']"/>
+
 </xsl:stylesheet>
