diff --git a/publish/generic-preview/oscal_catalog_html.xsl b/publish/generic-preview/oscal_catalog_html.xsl
index fedc9c0..e15bbb6 100644
--- a/publish/generic-preview/oscal_catalog_html.xsl
+++ b/publish/generic-preview/oscal_catalog_html.xsl
@@ -48,7 +48,11 @@
    
    <xsl:template match="catalog">
       <xsl:apply-templates select="metadata/title"/>
-      <xsl:apply-templates mode="metadata" select="metadata"/>
+      <div class="form-check form-switch">
+         <label class="form-check-label" for="requiredControlsSwitch">Show Required Controls Only</label>
+         <input class="form-check-input" type="checkbox" role="switch" id="requiredControlsSwitch"/>
+      </div>
+<!--      <xsl:apply-templates mode="metadata" select="metadata"/> -->
       <xsl:call-template name="table-of-contents"/>
       <main class="catalog">
          <xsl:apply-templates/>
@@ -99,8 +103,19 @@
    </xsl:template>
    
    <xsl:template match="control" mode="toc-listing">
-      <p class="toc-listing">
-         <xsl:apply-templates mode="#current" select="prop[@name='label'][1]"/>
+      <xsl:variable name="required">
+         <xsl:choose>
+            <xsl:when test="exists(prop[@name = 'tx_required_by']) or exists(prop[@name = 'tamus_required_by'])">
+               <xsl:text>required</xsl:text>
+            </xsl:when>
+            <xsl:otherwise>
+               <xsl:text>not_required</xsl:text>
+            </xsl:otherwise>
+         </xsl:choose>
+      </xsl:variable>
+
+      <p class="toc-listing { $required }">
+         <xsl:apply-templates mode="#current" select="prop[@name='label'][2]"/>
          <xsl:text> </xsl:text>
          <xsl:apply-templates mode="#current" select="title"/>
       </p>
@@ -140,7 +155,7 @@
    </xsl:template>
    
    <xsl:template priority="2" match="control/control/prop[@name='label']" mode="toc-listing">
-      <xsl:variable name="parent-label" select="../parent::control/prop[@name='label'][1]/@value"/>
+      <xsl:variable name="parent-label" select="../parent::control/prop[@name='label'][2]/@value"/>
       <span class="toc-label">
          <xsl:value-of select="substring-after(@value,$parent-label)"/>
       </span>
@@ -423,7 +438,7 @@
    </xsl:template>
    
    <xsl:template match="control" mode="link-as-link" expand-text="true">
-      <a href="#{ @id }">{ prop[@name='label'][1]/@value }</a>
+      <a href="#{ @id }">{ prop[@name='label'][2]/@value }</a>
    </xsl:template>
    
    <xsl:template priority="2" match="resource[empty(rlink)]" mode="link-as-link">
@@ -497,5 +512,4 @@
       </p>
    </xsl:template>
    
-   
 </xsl:stylesheet>
