--- publish/nist-emulation/sp800-53-catalog_html.xsl	2023-07-05 22:08:26
+++ publish/nist-emulation/sp800-53-catalog_html.xsl	2023-07-03 16:49:35
@@ -237,7 +237,7 @@
    </xsl:template>
    
    <xsl:template priority="3" match="control/part[@name='objective']">
-      <div class="part">
+      <div class="part {@name}">
          <xsl:copy-of select="@id"/>
          <details>
             <summary>
@@ -259,8 +259,9 @@
       </div>
    </xsl:template>
    
+<!-- DISABLED
    <xsl:template priority="3" match="control/part[@name='assessment-objective']">
-      <div class="part">
+      <div class="part {@name}">
          <xsl:copy-of select="@id"/>
          <details>
             <summary>
@@ -274,7 +275,8 @@
          </details>
       </div>
    </xsl:template>
-   
+-->
+
    <xsl:template priority="5" mode="assessment-table" match="part[@name='assessment-objective']//part[@name='assessment-objective']">
       <xsl:apply-templates mode="#current"/>
    </xsl:template>
@@ -733,5 +735,9 @@
          </xsl:otherwise>
       </xsl:choose>
    </xsl:template>
-   
+
+   <xsl:template priority="3" match="control/part[@name='assessment-objective']"/>
+
+   <xsl:template priority="3" match="control/part[@name='assessment-method']"/>
+
 </xsl:stylesheet>
