diff --git a/publish/nist-emulation/sp800-53A-catalog_html.xsl b/publish/nist-emulation/sp800-53A-catalog_html.xsl
index 1b7dfe9..f334665 100644
--- a/publish/nist-emulation/sp800-53A-catalog_html.xsl
+++ b/publish/nist-emulation/sp800-53A-catalog_html.xsl
@@ -40,7 +40,17 @@
    
    <xsl:template match="control">
       <xsl:variable name="withdrawn" select="some $p in (prop[@name='status']) satisfies matches($p/@value,'Withdrawn','i')"/>
-      <div class="control{ $withdrawn[boolean(.)] ! ' withdrawn' }">
+      <xsl:variable name="required">
+         <xsl:choose>
+            <xsl:when test="exists(prop[@name = 'tx_required_by']) or exists(prop[@name = 'tamus_required_by'])">
+               <xsl:text>required</xsl:text>
+            </xsl:when>
+            <xsl:otherwise>
+               <xsl:text>not_required</xsl:text>
+            </xsl:otherwise>
+         </xsl:choose>
+      </xsl:variable>
+      <div class="control{ $withdrawn[boolean(.)] ! ' withdrawn' } { $required }">
          <xsl:copy-of select="@id"/>
          <details>
             <!-- Keeping control enhancements closed -->
@@ -93,7 +103,7 @@
    
    <xsl:template name="control-label">
       <span class="label">
-         <xsl:apply-templates mode="part-number" select="prop[@name = 'label'][1]"/>
+         <xsl:apply-templates mode="part-number" select="prop[@name = 'label'][2]"/>
       </span>
    </xsl:template>
    
@@ -118,9 +128,6 @@
                   select="if ($count-of-things = 1) then $sg else $pl"/>
             </span>
             <xsl:text> </xsl:text>
-            <span class="count">
-               <xsl:value-of select="if ($count-of-things eq 0) then ' None' else $count-of-things"/>
-            </span>
          </summary>
          <xsl:apply-templates select="$things"/>
       </details>
@@ -386,7 +393,7 @@
          <xsl:variable name="withdrawn-to" select="../link[@rel = ('moved-to', 'incorporated-into')]"/>
          <xsl:variable name="link-label">
             <xsl:choose>
-               <xsl:when test="empty($withdrawn-to)">. </xsl:when>
+               <xsl:when test="empty($withdrawn-to)">.</xsl:when>
                <xsl:when test="$withdrawn-to/@rel = 'incorporated-into'">: Incorporated into </xsl:when>
                <xsl:otherwise>: Moved to </xsl:otherwise>
             </xsl:choose>
@@ -611,4 +618,151 @@
       </a>
    </xsl:template>
    
+   <!-- TAMUS SPECIFIC -->
+   <xsl:template match="prop[@name='implementation-level']">
+      <p class="prop {@name}">
+         <span class="inline-head">Implementation Level</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="concat(upper-case(substring(@value,1,1)),substring(@value,2))"/>
+      </p>
+   </xsl:template>
+
+   <xsl:template match="prop[@name='contributes-to-assurance']">
+      <p class="prop {@name}">
+         <span class="inline-head">Contributes to Assurance</span>
+         <xsl:text>: </xsl:text>
+         <xsl:choose>
+            <xsl:when test="@value = 'true'">Yes</xsl:when>
+            <xsl:otherwise>No</xsl:otherwise>
+         </xsl:choose>
+      </p>
+   </xsl:template>
+
+   <xsl:template match="prop[@name='tx_baseline']">
+      <p class="prop {@name}">
+         <span class="inline-head">Texas DIR Baseline</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="@value"/>
+      </p>
+   </xsl:template>
+
+   <xsl:template match="prop[@name='tx_privacy_baseline']">
+      <p class="prop {@name}">
+         <span class="inline-head">Texas DIR Privacy Baseline</span>
+         <xsl:text>: </xsl:text>
+         <xsl:choose>
+            <xsl:when test="@value = 'true'">Yes</xsl:when>
+            <xsl:otherwise>No</xsl:otherwise>
+         </xsl:choose>
+      </p>
+   </xsl:template>
+
+   <xsl:template match="prop[@name='tx_required_by']">
+      <p class="prop {@name}">
+         <span class="inline-head">Texas DIR Required By</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="@value"/>
+      </p>
+   </xsl:template>
+
+   <xsl:template match="prop[@name='tamus_required_by']">
+      <p class="prop {@name}">
+         <span class="inline-head">Texas A&amp;M System Required By</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="@value"/>
+      </p>
+   </xsl:template>
+
+   <xsl:template match="prop[@name='tx_new_requirement']">
+      <p class="prop {@name}">
+         <span class="inline-head">Texas DIR New Requirement</span>
+         <xsl:text>: </xsl:text>
+         <xsl:choose>
+            <xsl:when test="@value = 'true'">Yes</xsl:when>
+            <xsl:otherwise>No</xsl:otherwise>
+         </xsl:choose>
+      </p>
+   </xsl:template>
+
+   <xsl:template match="prop[@name='tamus_new_requirement']">
+      <p class="prop {@name}">
+         <span class="inline-head">Texas A&amp;M System New Requirement</span>
+         <xsl:text>: </xsl:text>
+         <xsl:choose>
+            <xsl:when test="@value = 'true'">Yes</xsl:when>
+            <xsl:otherwise>No</xsl:otherwise>
+         </xsl:choose>
+      </p>
+   </xsl:template>
+
+   <xsl:template match="control/part[@name='tx_implementation']" priority="2">
+      <div class="part statement {@name}">
+         <xsl:copy-of select="@id"/>
+            <xsl:apply-templates/>
+       </div>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/part[@name='tx_implementation'][empty(child::p)]">
+      <div class="part statement {@name}">
+         <xsl:copy-of select="@id"/>
+         <p><span class="inline-head">Texas DIR Implementation</span>:</p>
+         <xsl:apply-templates/>
+      </div>
+   </xsl:template>
+   
+<!-- Statements of withdrawn controls are picked up with the 'withdrawn' prop below. -->
+   <xsl:template  priority="5" match="control[matches(prop[@name='status'][1]/@value,'Withdrawn','i')]/part[@name='tx_implementation']"/>
+   
+   <xsl:template match="control/part[@name='tx_implementation']/p">
+      <xsl:choose>
+         <xsl:when test=". is ancestor::part[@name='tx_implementation']/child::p[1]">
+            <p><span class="inline-head">Texas DIR Implementation</span>
+               <xsl:text>: </xsl:text>
+               <xsl:apply-templates/></p>
+         </xsl:when>
+         <xsl:otherwise>
+            <xsl:next-match/>
+         </xsl:otherwise>
+      </xsl:choose>
+   </xsl:template>
+
+   <xsl:template priority="2" match="part[@name='tamus_implementation']" mode="title">
+      <h4>Texas A&amp;M System Implementation</h4>
+   </xsl:template>
+   
+   <xsl:template match="control/part[@name='tamus_implementation']" priority="2">
+      <div class="part statement {@name}">
+         <xsl:copy-of select="@id"/>
+            <xsl:apply-templates/>
+       </div>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/part[@name='tamus_implementation'][empty(child::p)]">
+      <div class="part statement {@name}">
+         <xsl:copy-of select="@id"/>
+         <p><span class="inline-head">Texas A&amp;M System Implementation</span>:</p>
+         <xsl:apply-templates/>
+      </div>
+   </xsl:template>
+   
+<!-- Statements of withdrawn controls are picked up with the 'withdrawn' prop below. -->
+   <xsl:template  priority="5" match="control[matches(prop[@name='status'][1]/@value,'Withdrawn','i')]/part[@name='tamus_implementation']"/>
+   
+   <xsl:template match="control/part[@name='tamus_implementation']/p">
+      <xsl:choose>
+         <xsl:when test=". is ancestor::part[@name='tamus_implementation']/child::p[1]">
+            <p><span class="inline-head">Texas A&amp;M System Implementation</span>
+               <xsl:text>: </xsl:text>
+               <xsl:apply-templates/></p>
+         </xsl:when>
+         <xsl:otherwise>
+            <xsl:next-match/>
+         </xsl:otherwise>
+      </xsl:choose>
+   </xsl:template>
+
+   <xsl:template priority="2" match="part[@name='tamus_implementation']" mode="title">
+      <h4>Texas A&amp;M System Implementation</h4>
+   </xsl:template>
+
 </xsl:stylesheet>
