diff --git a/publish/nist-emulation/sp800-53A-catalog_html.xsl b/publish/nist-emulation/sp800-53A-catalog_html.xsl
index 1b7dfe9..529d908 100644
--- a/publish/nist-emulation/sp800-53A-catalog_html.xsl
+++ b/publish/nist-emulation/sp800-53A-catalog_html.xsl
@@ -610,5 +610,197 @@
          <xsl:value-of select="prop[@name='label']/@value"/>
       </a>
    </xsl:template>
+
+<!-- TAMUS ADDED CONTENT BELOW HERE -->
+
+  <xsl:template priority="1" match="metadata" mode="metadata">
+      <section class="metadata">
+         <details>
+            <summary><span class="lbl">OSCAL Catalog Metadata</span></summary>
+            <xsl:apply-templates select="published, last-modified, version, oscal-version, doc-id, prop, link"/>
+            <!--
+            <xsl:for-each-group select="role" group-by="true()">
+               <div class="block roles">
+                  <xsl:apply-templates select="current-group()"/>
+               </div>
+            </xsl:for-each-group>
+            <xsl:for-each-group select="party" group-by="true()">
+               <div class="block parties">
+                  <xsl:apply-templates select="current-group()"/>
+               </div>
+            </xsl:for-each-group>
+            -->
+            <!-- responsibile-parties elements bind parties to roles, handled with parties -->
+            <xsl:apply-templates select="remarks"/>
+         </details>
+      </section>
+   </xsl:template>
+
+   <xsl:template priority="1" match="published">
+      <p><span class="inline-head">Published Date</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="."/>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="1" match="last-modified">
+      <p><span class="inline-head">Last Substantively Modified Date</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="."/>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="1" match="version">
+      <p><span class="inline-head">Version</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="."/>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="1" match="oscal-version"/>
+
+   <xsl:template priority="1" match="role"/>
+
+   <xsl:template priority="1" match="party"/>
+
+   <xsl:template priority="3" match="control/prop[@name='implementation-level']">
+      <xsl:copy-of select="@id"/>
+      <p><span class="inline-head">Implementation Level</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="concat(upper-case(substring(@value,1,1)),substring(@value,2))"/>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/prop[@name='contributes-to-assurance']">
+      <xsl:copy-of select="@id"/>
+      <p><span class="inline-head">Contributes to Assurance</span>
+         <xsl:text>: </xsl:text>
+         <xsl:choose>
+            <xsl:when test="@value eq 'true'">Yes</xsl:when>
+            <xsl:otherwise>No</xsl:otherwise>
+         </xsl:choose>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/prop[@name='tx_baseline']">
+      <xsl:copy-of select="@id"/>
+      <p><span class="inline-head">Texas DIR Baseline</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="@value"/>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/prop[@name='tx_privacy_baseline']">
+      <xsl:copy-of select="@id"/>
+      <p><span class="inline-head">Texas DIR Privacy Baseline</span>
+         <xsl:text>: </xsl:text>
+         <xsl:choose>
+            <xsl:when test="@value eq 'true'">Yes</xsl:when>
+            <xsl:otherwise>No</xsl:otherwise>
+         </xsl:choose>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/prop[@name='tamus_baseline']">
+      <xsl:copy-of select="@id"/>
+      <p><span class="inline-head">Texas A&amp;M System Baseline</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="@value"/>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/prop[@name='tx_new_requirement']">
+      <xsl:copy-of select="@id"/>
+      <p><span class="inline-head">Texas DIR New Requirement</span>
+         <xsl:text>: </xsl:text>
+         <xsl:choose>
+            <xsl:when test="@value eq 'true'">Yes</xsl:when>
+            <xsl:otherwise>No</xsl:otherwise>
+         </xsl:choose>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/prop[@name='tamus_new_requirement']">
+      <xsl:copy-of select="@id"/>
+      <p><span class="inline-head">Texas A&amp;M System New Requirement</span>
+         <xsl:text>: </xsl:text>
+         <xsl:choose>
+            <xsl:when test="@value eq 'true'">Yes</xsl:when>
+            <xsl:otherwise>No</xsl:otherwise>
+         </xsl:choose>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/prop[@name='tx_required_by']">
+      <xsl:copy-of select="@id"/>
+      <p><span class="inline-head">Texas DIR Required By</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="@value"/>
+      </p>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/prop[@name='tamus_required_by']">
+      <xsl:copy-of select="@id"/>
+      <p><span class="inline-head">Texas A&amp;M System Required By</span>
+         <xsl:text>: </xsl:text>
+         <xsl:value-of select="@value"/>
+      </p>
+   </xsl:template>
+   
+   <xsl:template match="control/part[@name='tx_implementation']" priority="2">
+      <div class="part statement {@name}">
+         <xsl:copy-of select="@id"/>
+            <xsl:apply-templates/>
+       </div>
+   </xsl:template>
+
+   <xsl:template match="control/part[@name='tamus_implementation']" priority="2">
+      <div class="part statement {@name}">
+         <xsl:copy-of select="@id"/>
+            <xsl:apply-templates/>
+       </div>
+   </xsl:template>
+
+   <xsl:template priority="3" match="control/part[@name='tx_implementation'][empty(child::p)]">
+      <div class="part statement {@name}">
+         <xsl:copy-of select="@id"/>
+         <p><span class="inline-head">Texas DIR Implementation Statement</span>:</p>
+         <xsl:apply-templates/>
+      </div>
+   </xsl:template>
+   
+   <xsl:template priority="3" match="control/part[@name='tamus_implementation'][empty(child::p)]">
+      <div class="part statement {@name}">
+         <xsl:copy-of select="@id"/>
+         <p><span class="inline-head">Texas A&amp;M System Implementation Statement</span>:</p>
+         <xsl:apply-templates/>
+      </div>
+   </xsl:template>
+   
+   <xsl:template match="control/part[@name='tx_implementation']/p">
+      <xsl:choose>
+         <xsl:when test=". is ancestor::part[@name='tx_implementation']/child::p[1]">
+            <p><span class="inline-head">Texas DIR Implementation Statement</span>
+               <xsl:text>: </xsl:text>
+               <xsl:apply-templates/></p>
+         </xsl:when>
+         <xsl:otherwise>
+            <xsl:next-match/>
+         </xsl:otherwise>
+      </xsl:choose>
+   </xsl:template>
+   
+   <xsl:template match="control/part[@name='tamus_implementation']/p">
+      <xsl:choose>
+         <xsl:when test=". is ancestor::part[@name='tamus_implementation']/child::p[1]">
+            <p><span class="inline-head">Texas A&amp;M System Implementation Statement</span>
+               <xsl:text>: </xsl:text>
+               <xsl:apply-templates/></p>
+         </xsl:when>
+         <xsl:otherwise>
+            <xsl:next-match/>
+         </xsl:otherwise>
+      </xsl:choose>
+   </xsl:template>
    
 </xsl:stylesheet>
